api/v1alpha1/common_types.go

```diff
@@ -0,0 +1,37 @@
+package v1alpha1
+
+import corev1 "k8s.io/api/core/v1"
+
+// Autoscaling configuration for scalable workloads.
+type Autoscaling struct {
```

Comment:
Are we planning to always deploy HPA or is this type intended to abstract whatever scaling is used?

Reply:
HPA is the standard way to scale Pods in K8s. It supports standard metrics like cpu/metrics as well as policies that support custom metrics if we ever want to invest in that.

Reply:
It sounds like the answer is yes, then. Would it make sense to just use the HPAv2 Spec type here instead of defining our own?

Reply:
We have the need for both short and long-lived cluster executions. In the former case, HPA will most likely not be enabled, whereas the latter seems more likely to want HPA.

I think reworking this to leverage `minReplicas` might make more sense but I don't feel inclined to expose the entire resource. The idea behind this sentiment is, if we understand how ray clusters work well, then we should be able to abstract some scaling details and expose only the fields that are relevant to different workloads.